### PR TITLE
rewrite SBOM handling code

### PR DESCRIPTION
**Pull request by matthyx**
## PR Type:
Refactoring, Enhancement, Tests

___
## PR Description:
This pull request includes a significant refactoring of the SBOM (Software Bill of Materials) handling code. It introduces a new `SBOMHandler` struct in the `pkg/sbomhandler/v1` package, which is responsible for filtering SBOMs and managing image usage counts. The `SBOMHandler` interacts with a `StorageClient` to retrieve and store SBOM data. The PR also includes tests for the new `SBOMHandler` functionality.

Additionally, the PR modifies the `FileHandler` interface and its implementations, changing the `GetFiles` method to `GetAndDeleteFiles`, reflecting the method's behavior more accurately.

The PR also includes the deletion of several files related to the old SBOM handling implementation, indicating a shift towards the new structure.

___
## PR Main Files Walkthrough:
`pkg/sbomhandler/v1/sbomhandler.go`: Introduces the `SBOMHandler` struct and its methods. The `SBOMHandler` is responsible for filtering SBOMs based on relevancy and managing image usage counts. It interacts with a `StorageClient` to retrieve and store SBOM data.
`pkg/sbomhandler/v1/sbomhandler_test.go`: Contains unit tests for the `SBOMHandler` methods, including `FilterSBOM`, `IncrementImageUse`, and `DecrementImageUse`.
`pkg/filehandler/v1/filehandler_test.go`: Modifies existing tests to accommodate the change from `GetFiles` to `GetAndDeleteFiles` in the `FileHandler` interface.
`pkg/storage/v1/storage_test.go`: Updates tests to reflect changes in the SBOM handling code.
`pkg/storage/v1/storage_nocache.go`: Updates the `StorageClient` implementation to reflect changes in the SBOM handling code.
`pkg/storage/storage_mock.go`: Updates the mock `StorageClient` used for testing to reflect changes in the SBOM handling code.
`main.go`: Updates the main application code to use the new `SBOMHandler` instead of the old SBOM handling code.
`pkg/relevancymanager/v1/relevancy_manager.go`: Updates the `RelevancyManager` code to use the new `SBOMHandler`.
`pkg/relevancymanager/v1/relevancy_manager_test.go`: Updates tests for the `RelevancyManager` to use the new `SBOMHandler`.
`pkg/containerwatcher/v1/container_watcher.go`: Updates the `ContainerWatcher` code to use the new `SBOMHandler`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,11 +7,11 @@ require (
 	github.com/cilium/ebpf v0.11.0
 	github.com/facette/natsort v0.0.0-20181210072756-2cd4dd1e2dcb
 	github.com/inspektor-gadget/inspektor-gadget v0.18.2-0.20230802150404-2f6b36dd8642
```
Comment (dwertent) [inline go.mod:9]: Consider updating to `v0.19`

Comment (matthyx) [reply go.mod:9]: ok bumping deps...

Comment (matthyx) [reply go.mod]: testing if it still works...

File: pkg/storage/v1/storage_nocache.go [hunk 2]
```diff
@@ -70,12 +70,6 @@ func (sc StorageNoCache) GetSBOM(name string) (*v1beta1.SBOMSPDXv2p3, error) {
 		return nil, err
 	}
 
-	// remove fields that are not needed
```
Comment (matthyx) [inline pkg/storage/v1/storage_nocache.go]: I will never again copy paste from old code during a refactoring... it's only good to forget things to cleanup later.

**Comment by matthyx** [comment]
> /improve

curious to see the result of this...

**Review by dwertent — APPROVED** [review]

**Review by dwertent — APPROVED** [review]

**Pull request merged**